### PR TITLE
feat: Rename Text stack to align with Design library

### PR DESCRIPTION
**Pull request by frankieyan**
## Short description

This updates the `Text` component to align its size and weight stack with the Doist product library. This will allow us to identify font-size tokens easier when working with Figma:

image

Reference: https://twist.com/a/1585/ch/414345/t/2200903/c/65916995
Figma: https://www.figma.com/file/LYlWNzvhMDh907l07mPPQk/Product-Web?node-id=239%3A26

Note that our stories have their font-stack set to Arial, which doesn't seem to support the `semibold`/`500` font-weight. In our products, they should look correct with the system fonts font-stack.

|Design|Before|After|
|-|-|-|
|image|image|image|

## PR Checklist

-   [ ] Added tests for bugs / new features
-   [x] Updated docs (storybooks, readme)
-   [x] Executed `npm run validate` and made sure no errors / warnings were shown
-   [x] Described changes in `CHANGELOG.md`
-   [x] Bumped version in `package.json` and `package-lock.json` (`npm --no-git-tag-version version <major|minor|patch>`) [ref](https://docs.npmjs.com/cli/v6/commands/npm-version)
-   [x] Updated all static build artifacts (`npm run build-all`)

## Versioning

v9.2.0-beta.10

### CODE REVIEW

**Comment by frankieyan** [comment]
cc. @gnapse and @Doist/frontend-product so you are aware

**Comment by pedroalves0** [comment]
@henningmu I suppose that component could have been used for example in https://github.com/Doist/todoist-web/pull/2983.

File: src/new-components/default-styles.less [hunk 1]
```diff
@@ -23,10 +23,10 @@
     --reactist-spacing-xxlarge: 32px;
 
     /* font sizes */
-    --reactist-font-size-xsmall: 10px;
-    --reactist-font-size-small: 12px;
-    --reactist-font-size-normal: 14px;
-    --reactist-font-size-large: 16px;
+    --reactist-font-size-caption: 12px;
```
Comment (craigcarlyle) [inline src/new-components/default-styles.less:26]: I love these purpose-based names ❤️

**Review by craigcarlyle — APPROVED** [review]
🎉

**Comment by frankieyan** [comment]
> @henningmu I suppose that this component could have been used for example in [Doist/todoist-web#2983](https://github.com/Doist/todoist-web/pull/2983).

Not until we can get rid of IE11 over there since these make use of CSS variables 👀

**Pull request merged**